### ISSUE TITLE
Only require `key` for adjacent sibling widgets

### ISSUE THREAD

**Issue by bitpshr** [dojo/widget-core#661]
**Enhancement**

It would be nice if only adjacent sibling widgets using the same class required keys.

Package Version: latest

**Code**

**Expected behavior:**

```ts
render() {
    return v('div', {}, [
        v('div', {}, [ w(MyWidget, {})]),
        v('div', {}, [ w(MyWidget, {})])
    ]);
}
```

**Actual behavior:**

```ts
render() {
    return v('div', {}, [
        v('div', {}, [ w(MyWidget, { key: 'a' })]),
        v('div', {}, [ w(MyWidget, { key: 'b' })])
    ]);
}
```

**Comment by kitsonk** [comment]
What happens/occurs when you omit the key?

`key` is becoming increasingly required for things like meta providers.  Doing the analysis of if they are truly required or not might be expensive.

**Comment by kitsonk** [comment]
This can be addressed now with the landing of #725.

Removing discussion and scheduling for beta5.

**Comment by agubler** [comment]
@bitpshr The key check should now only be required for siblings.
